File: Baekjoon_14888.kt
private var numCnt = 0
private val nums = mutableListOf<Int>()
private val operators = IntArray(4)
private var max = Int.MIN_VALUE
private var min = Int.MAX_VALUE

fun main() {
    input()
    calculation(nums[0], 1)
    output()
}

private fun input() {
    numCnt = readln().toInt()
    readln().split(" ").map { it.toInt() }.forEach { nums.add(it) }
    readln().split(" ").map { it.toInt() }.forEachIndexed { index, cnt ->
        operators[index] = cnt
    }
}

private fun calculation(total: Int, idx: Int) {
    if(idx == numCnt) {
        if(total < min) min = total
        if(total > max) max = total

        return
    }

    for(i in 0 until 4) {
        if(operators[i] > 0) {
            operators[i]--

            when(i) {
                0 -> calculation(total + nums[idx], idx + 1)
                1 -> calculation(total - nums[idx], idx + 1)
                2 -> calculation(total * nums[idx], idx + 1)
                3 -> calculation(total / nums[idx], idx + 1)
            }

            operators[i]++
        }
    }
}

private fun output() {
    println(max)
    println(min)
}

File: Baekjoon_14888.cs
using System;
using System.Linq;

class Baekjoon_14888
{
    static int[] numbers;
    static int[] operators;
    static int max = int.MinValue;
    static int min = int.MaxValue;

    static void Main(string[] args)
    {
        int N = int.Parse(Console.ReadLine());
        numbers = Console.ReadLine().Split().Select(int.Parse).ToArray();
        operators = Console.ReadLine().Split().Select(int.Parse).ToArray();

        Solution(numbers[0], 1);

        Console.WriteLine(max);
        Console.WriteLine(min);
    }

    static void Solution(int current, int index)
    {
        if (index == numbers.Length)
        {
            max = Math.Max(max, current);
            min = Math.Min(min, current);
            return;
        }

        for (int i = 0; i < 4; i++)
        {
            if (operators[i] > 0)
            {
                operators[i]--;
                int next = Calculate(current, numbers[index], i);
                Solution(next, index + 1);
                operators[i]++;
            }
        }
    }

    static int Calculate(int a, int b, int op)
    {
        switch (op)
        {
            case 0: return a + b;
            case 1: return a - b;
            case 2: return a * b;
            case 3: return a / b;
            default: throw new ArgumentException("Invalid operator");
        }
    }
}